src/averageAges.js

```diff
@@ -36,8 +41,19 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  let females = people.filter(person => person.sex === 'f');
+
+  females = withChildren
+    ? females.filter(person => people
+      .map(child => child.mother)
+      .filter(women => women).includes(person.name))
+    : females;
+
+  const ageSum = females
+    .reduce((sum, person) => sum + (person.died - person.born), 0);
```

Comment:
Pay attention to the formatting 

<img width="383" alt="Screenshot 2023-03-23 at 17 16 31" src="https://user-images.githubusercontent.com/104769168/227249404-32d1e8fd-6dc1-44dd-ba0c-56f6cee14f76.png">